blog/02-18-2023-Build-an-NFT-Full-Stack-Monster-War-Game-on-The-Celo-Blockchain/Build-an-NFT-Full-Stack-Monster-War-Game-on-The-Celo-Blockchain.md

```diff
@@ -0,0 +1,998 @@
+  } catch (e) {
+    console.log({ e });
+  }
+};
+
+export const minted = async (minterContract, _address) => {
+  try {
+    return await minterContract.methods.hasMinted(_address).call();
+  } catch (e) {
+    console.log({ e });
+  }
+};
+
+export const checkPowervalue = async (minterContract, _address, _index) => {
+  try {
+    return await minterContract.methods.canSwallowNFT(_address, _index).call();
+  } catch (e) {
+    console.log({ e });
+  }
+};
+
+// get the address that deployed the NFT contract
+export const fetchNftContractOwner = async (minterContract) => {
+  try {
+    let owner = await minterContract.methods.owner().call();
+    return owner;
+  } catch (e) {
+    console.log({ e });
+  }
+};
+
+export const swallow = async (minterContract, performActions, index) => {
+  try {
+    await performActions(async (kit) => {
+      try {
+        console.log(minterContract, index);
+        const { defaultAccount } = kit;
+        await minterContract.methods
+          .swallowNFT(index)
+          .send({ from: defaultAccount });
+      } catch (error) {
+        console.log({ error });
+      }
+    });
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+export const upgrade = async (minterContract, performActions, index) => {
+  try {
+    await performActions(async (kit) => {
+      try {
+        const price = ethers.utils.parseUnits(String(0.5), "ether");
+        const { defaultAccount } = kit;
+        await minterContract.methods
+          .upgradeNFT(index)
+          .send({ from: defaultAccount, value: price });
+      } catch (error) {
+        console.log({ error });
+      }
+    });
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+export const remove = async (minterContract, performActions, index) => {
+  try {
+    await performActions(async (kit) => {
+      try {
+        const { defaultAccount } = kit;
+        await minterContract.methods
+          .remove(index)
+          .send({ from: defaultAccount });
+      } catch (error) {
+        console.log({ error });
+      }
+    });
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+```
+
+Now lets break down the components
+
+```javascript
+import { ethers } from "ethers";
+
+// mint an NFT
+export const createNft = async (minterContract, performActions, name) => {
+  await performActions(async (kit) => {
+    if (!name) return;
+    const { defaultAccount } = kit;
+
+    try {
+      let transaction = await minterContract.methods
+        .mint(name)
+        .send({ from: defaultAccount });
+
+      return transaction;
+    } catch (error) {
+      console.log("Error uploading file: ", error);
+    }
+  });
+};
+```
+
+First, we made the necessary import and then we declared the `createNft()` function. This function is used to mint an NFT. It takes the `minterContract` and `performActions` as arguments, along with the name of the NFT. It then uses the `defaultAccount` from the kit to mint the NFT. 
```

Comment:
First, we made the necessary import and then declared the `createNft()` function. This function is used to mint an NFT. It takes the `minterContract` and `performActions` as arguments, along with the name of the NFT. It then uses the `defaultAccount` from the kit to mint the NFT.